internal/services/netapp/netapp_volume_group_sap_hana_resource.go

```diff
@@ -0,0 +1,628 @@
+				if volumeCrr.Properties.DataProtection != nil &&
+					volumeCrr.Properties.DataProtection.Replication != nil &&
+					strings.EqualFold(string(*volumeCrr.Properties.DataProtection.Replication.EndpointType), string(volumegroups.EndpointTypeDst)) {
+
+					// Modify volumeType as data protection type on main volumeList
+					// so it gets created correctly as data protection volume
+					(*volumeList)[i].Properties.VolumeType = utils.String("DataProtection")
+				}
+			}
+
+			// TODO: deploymentSpecId is temporary until the backend is updated and deploymentSpecId is not required anymore,
+			//       it will be handled internally by the RP
+			deploymentSpecId := "20542149-bfca-5618-1879-9863dc6767f1"
+
+			parameters := volumegroups.VolumeGroupDetails{
+				Location: utils.String(location.Normalize(model.Location)),
+				Properties: &volumegroups.VolumeGroupProperties{
+					GroupMetaData: &volumegroups.VolumeGroupMetaData{
+						GroupDescription:      utils.String(model.GroupDescription),
+						ApplicationType:       &applicationType,
+						ApplicationIdentifier: utils.String(model.ApplicationIdentifier),
+						DeploymentSpecId:      utils.String(deploymentSpecId),
+					},
+					Volumes: volumeList,
+				},
+			}
+
+			err = client.VolumeGroupsCreateThenPoll(ctx, id, parameters)
+			if err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			// Waiting for volume group be completely provisioned
+			if err := waitForVolumeGroupCreateOrUpdate(ctx, client, id); err != nil {
+				return err
+			}
+
+			// CRR - Authorizing secondaries from primary volumes
+			for _, volumeCrr := range *volumeList {
+				if volumeCrr.Properties.DataProtection != nil &&
+					volumeCrr.Properties.DataProtection.Replication != nil &&
+					strings.EqualFold(string(*volumeCrr.Properties.DataProtection.Replication.EndpointType), string(volumegroups.EndpointTypeDst)) {
+
+					// Getting secondary volume resource id
+					secondaryId := volumes.NewVolumeID(subscriptionId,
+						model.ResourceGroupName,
+						model.AccountName,
+						getResourceNameString(volumeCrr.Properties.CapacityPoolResourceId),
+						getResourceNameString(volumeCrr.Name),
+					)
+
+					// Getting primary resource id
+					primaryId, err := volumesreplication.ParseVolumeID(volumeCrr.Properties.DataProtection.Replication.RemoteVolumeResourceId)
+					if err != nil {
+						return err
+					}
+
+					// Authorizing
+					if err = replicationClient.VolumesAuthorizeReplicationThenPoll(ctx, *primaryId, volumesreplication.AuthorizeRequest{
+						RemoteVolumeResourceId: utils.String(secondaryId.ID()),
+					},
+					); err != nil {
+						return fmt.Errorf("cannot authorize volume replication: %v", err)
+					}
+
+					// Wait for volume replication authorization to complete
+					log.Printf("[DEBUG] Waiting for replication authorization on %s to complete", id)
+					if err := waitForReplAuthorization(ctx, replicationClient, *primaryId); err != nil {
```

Comment:
Similar to CreateThenPoll, `VolumesAuthorizeReplicationThenPoll` should be enough and if not, we should open a bug on pandora.

Reply:
As I explained in the other response, we cannot get rid of this since our backend returns that the resource is created prematurely so if you don't have a GET loop to double check this, then failures will happen.